src/scriptInterfaceMagic.h

```diff
@@ -216,80 +203,91 @@ template<typename T> struct convert<sf::Vector3<T> >
         v.z = luaL_checknumber(L, idx++);
     }
 };
-
+
 /* Convert parameters to std::vector<?> objects. */
 template<typename T> struct convert<std::vector<T> >
 {
     static void param(lua_State* L, int& idx, std::vector<T>& v)
-    {
-        while(idx <= lua_gettop(L))
-        {
-            T var;
+    {
+        while(idx <= lua_gettop(L))
+        {
+            T var;
             convert<T>::param(L, idx, var);
-            v.push_back(var);
-        }
-    }
-
-    static int returnType(lua_State* L, std::vector<T> vector)
-    {
-        lua_newtable(L);
-        int n = 1;
-        for(auto val : vector)
-        {
-            lua_pushnumber(L, n);
-            if (convert<T>::returnType(L, val))
-            {
-                lua_settable(L, -3);
-                n++;
-            }else{
-                lua_pop(L, 1);
-            }
-        }
-        return 1;
-    }
-};
-/* Convert parameters to std::map<string, ?> objects. */
-template<typename T> struct convert<std::map<string, T> >
-{
-    static int returnType(lua_State* L, std::map<string, T> map)
-    {
-        lua_newtable(L);
-        for(const auto& kv : map)
-        {
-            lua_pushstring(L, kv.first.c_str());
-            if (convert<T>::returnType(L, kv.second))
-            {
-                lua_settable(L, -3);
-            }else{
-                lua_pop(L, 1);
-            }
-        }
-        return 1;
-    }
-};
-
-template<class T, typename FuncProto> struct call
-{
-};
-
-template<class T> struct call<T, void(T::*)() >
-{
-    typedef void(T::*FuncProto)();
-    
-    static int function(lua_State* L)
-    {
-        FuncProto* func_ptr = reinterpret_cast<FuncProto*>(lua_touserdata(L, lua_upvalueindex (1)));
-        FuncProto func = *func_ptr;
-        T* obj = NULL;
-        int idx = 1;
+            v.push_back(var);
+        }
+    }
+
+    static int returnType(lua_State* L, std::vector<T> vector)
+    {
+        lua_newtable(L);
+        int table_idx = lua_gettop(L);
+        int n = 1;
+        for(auto val : vector)
+        {
+            int nr_vals = convert<T>::returnType(L, val);
+            if (nr_vals > 1)
+            {
+                LOG(WARNING) << "std::vector<" << typeid(val).name() << "> does not support types that return multiple values. Only the first one is used.";

```

Comment:
Error message looks like this:

´´´´
[WARN]  std::vector<N2sf7Vector2IfEE> does not support types that return multiple values. Only the first one is used.
´´´´